src/pysonar_scanner/configuration/cli.py

```diff
@@ -17,168 +17,24 @@
-            ("sonar.scanner.keystorePassword", scanner.keystore_password),
-            ("sonar.scanner.proxyHost", scanner.proxy_host),
-            ("sonar.scanner.proxyPort", scanner.proxy_port),
-            ("sonar.scanner.proxyUser", scanner.proxy_user),
-            ("sonar.scanner.proxyPassword", scanner.proxy_password),
-            ("sonar.scanner.wasJreCacheHit", scanner.was_jre_cache_hit.name if scanner.was_jre_cache_hit else None),
-            ("sonar.scanner.wasEngineCacheHit", scanner.was_engine_cache_hit),
-        ]
-
-        for key, value in optional_properties:
-            if value is not None and value != "":
-                properties.append({"key": key, "value": value})
-
-        return {"scannerProperties": properties}
-
-    def to_json(self) -> str:
-        return json.dumps(self.__to_dict(), indent=2)
-
-
-class ConfigurationLoader:
+class CliConfigurationLoader:
 
     @classmethod
-    def initialize_configuration(cls) -> Configuration:
+    def load(cls) -> dict[str, any]:
         args = cls.__parse_cli_args()
 
-        internal = Internal(args.sonar_scanner_internal_dump_to_file, args.sonar_scanner_internal_sq_version)
-
-        scanner = Scanner(
-            os=args.sonar_scanner_os,
-            arch=args.sonar_scanner_arch,
-            connect_timeout=args.sonar_scanner_connect_timeout,
-            socket_timeout=args.sonar_scanner_socket_timeout,
-            response_timeout=args.sonar_scanner_response_timeout,
-            truststore_path=args.sonar_scanner_truststore_path,
-            truststore_password=args.sonar_scanner_truststore_password,
-            keystore_path=args.sonar_scanner_keystore_path,
-            keystore_password=args.sonar_scanner_keystore_password,
-            proxy_host=args.sonar_scanner_proxy_host,
-            proxy_port=args.sonar_scanner_proxy_port,
-            proxy_user=args.sonar_scanner_proxy_user,
-            proxy_password=args.sonar_scanner_proxy_password,
-            skip_jre_provisioning=args.skip_jre_provisioning,
-            java_exe_path=args.sonar_scanner_java_exe_path,
-            java_opts=args.sonar_scanner_java_opts,
-            sonarcloud_url=args.sonar_scanner_cloud_url,
-            api_base_url=args.sonar_scanner_api_url,
-            internal=internal,
-        )
-
-        sonar = Sonar(
-            scanner=scanner,
-            verbose=args.verbose,
-            project_base_dir=args.sonar_project_base_dir,
-            user_home=args.sonar_user_home,
-            token=args.token,
-            project_key=args.sonar_project_key,
-            host_url=args.sonar_host_url,
-            region=args.sonar_region,
-        )
+        config = {}
+        for prop in properties.PROPERTIES:
+            if prop.cli_getter is not None:
+                value = prop.cli_getter(args)
+                config[prop.name] = value
 
-        return Configuration(sonar)
+        return {k: v for k, v in config.items() if v is not None}
```

Comment:
Super minor, and unrelated to the changes from this PR, but there is some inconsistent capitalization in the help strings under this:
- "Sonar scanner CLI for python" -> Python
- "Token used to authenticate against the SonarQube Server or SonarQube cloud" -> Cloud

I could be fixed in a different PR, but as I suspect you'll have to rebase, it would probably make sense to fix it at the same time.